### PR TITLE
[LED-122] Implemented US Bank QFX converter

### PR DESCRIPTION
**Pull request by taylerhow**
A few things:

- Modified the basic QFX Converter to derive the Type from the <TRNTYPE> data. DEBIT = Misc Debit & CREDIT = Misc Credit
- Implemented & tested the US Bank QFX Converter
- Modified the GenerateEpoch utility to handle epoch timestamps that do not include timezone info
- Added a couple cases to the TypeConversion.convert(String type) switch statement, for the US Bank QFX converter
- Added a sample US Bank QFX file to our test resources

Not important right now, but something to consider: Our QFX converters share some very similar code. It might be worthwhile to extract some of this code into an interface. Doesn't need to happen right now, but keep it in mind and we move forward and likely create several more converters for whatever file types Sean would like.

### CODE REVIEW

**Comment by LadyMopar** [comment]
When I import it file, it seems to populate the table just fine, but I get this error: 
image

File: src/main/java/ledger/io/input/QfxConverter.java [hunk 1]
```diff
@@ -120,7 +121,8 @@ private void parseXml(List<Transaction> transactions, Document xml) throws Conve
             for (int i = 0; i < transactionTypes.getLength(); i++) {
                 Date date = new Date(GenerateEpoch.generate(transactionDates.item(i).getTextContent()));
 
-                Type type = TypeConversion.convert("UNKNOWN");
+                Type type = TypeConversion.convert(transactionTypes.item(i).getTextContent());
```
Comment (shellajt) [inline src/main/java/ledger/io/input/QfxConverter.java]: Won't this throw an error if it is not one of our pre-defined types?

**Comment by shellajt** [comment]
@LadyMopar Sounds like you've got an old DB file again. Can you try making a new one.

**Comment by taylerhow** [comment]
@shellajt Unfortunately, the data provided in the various QFX fields differs quite a bit depending on the source. For the US Bank QFX converter, I had to be really specific in what columns I derived from what fields, etc. This converter will only work for US Bank.

**Comment by taylerhow** [comment]
@LadyMopar I think you might be using a DB file that's out of date (and doesn't have all of our current Types defined). Try making a new DB and importing into there.

**Comment by LadyMopar** [comment]
I've tried multiple times, but it doesn't seem to be working for me

**Comment by shellajt** [comment]
No need to close the pr...

**Comment by taylerhow** [comment]
@LadyMopar Fixed the error. Turns out we just we're adding one of our default Types to the DB upon creation 👎 I also renamed the QFX converters to their appropriate banks.

**Comment by shellajt** [comment]
@taylerhow I dont think that fixed it, i'm getting the same error now

**Comment by taylerhow** [comment]
@shellajt Did you make a new DB? All old DBs won't have the correct Types added

**Comment by shellajt** [comment]
I did, but forgot to pull. :-1:

**Comment by taylerhow** [comment]
👍

**Comment by shellajt** [comment]
@LadyMopar want to approve it then? :)

**Pull request merged**